dev/com.ibm.ws.jbatch.rest/src/com/ibm/ws/jbatch/rest/internal/resources/JobInstances.java

```diff
@@ -236,6 +236,13 @@ public void handleRequest(final RESTRequest request, final RESTResponse response
             response.sendError(HttpURLConnection.HTTP_UNAUTHORIZED, jse.getMessage());
         } catch (BatchNoSuchJobInstanceException bnsjie) {
             response.sendError(HttpURLConnection.HTTP_BAD_REQUEST, bnsjie.getMessage());
+        } catch (UnsupportedOperationException uoe) {
+        	
+        	// In case we decide to use UnsupportedOperationException for other reasons in the future...
+        	if(uoe.getMessage().contains("The REST URL search parameters requesting this function are not supported by the Java batch memory-based persistence configuration"))
+        		response.sendError(HttpURLConnection.HTTP_INTERNAL_ERROR, ResourceBundleRest.getMessage("in.memory.search.not.supported"));
```

Comment:
Since it's all in Open Liberty, should we just make a subclass instead of scraping the message?

Reply:
Modified solution, pushed new changes